internal/controller/apiextensions/claim/configurator.go

```diff
@@ -144,13 +141,15 @@ func (c *APIDryRunCompositeConfigurator) Configure(ctx context.Context, cm resou
 	ucp.SetClaimReference(proposed)
 
 	if !meta.WasCreated(cp) {
-		// The API server returns an available name derived from
-		// generateName when we perform a dry-run create. This name is
-		// likely (but not guaranteed) to be available when we create
-		// the composite resource. If the API server generates a name
-		// that is unavailable it will return a 500 ServerTimeout error.
 		cp.SetGenerateName(fmt.Sprintf("%s-", cm.GetName()))
-		return errors.Wrap(c.client.Create(ctx, cp, client.DryRunAll), errName)
+
+		// Generated name is likely (but not guaranteed) to be available
+		// when we create the composite resource. If taken,
+		// then we are going to update an existing composite,
+		// hijacking it from another claim. Depending on context/environment
+		// the consequences could be more or less serious.
+		// TODO: decide if we must prevent it.
```

Comment:
to be clear here: this is the same in the old code, @pedjak just made it clearer here in the comment.